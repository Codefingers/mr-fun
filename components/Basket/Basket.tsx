import React from 'react';
import styled from 'styled-components';

import BasketCard from '../BasketCard/BasketCard';
import Total from '../Total/Total';
import { useBasket } from '../../hooks/useBasket';
import { Product } from '../../types';

const BasketContainer = styled.div`
  margin: 16px 24px;
  padding: 16px 24px;
  margin-top: 68px;
`;

export const Basket = () => {
  const { basket, removeFromBasket } = useBasket();

  return (
    <BasketContainer>
      <h2>Your Basket</h2>
      <p>You're currently looking at purchasing the following passes:</p>
      <div>
        {basket?.map((item: Product, index) => {
          if (item && item.quantity) {
            return (
              <BasketCard
                item={item}
                key={`basket-item-${index}`}
                removeFromBasket={removeFromBasket}
              />
            );
          }
          return '';
        })}
      </div>
      <Total basket={basket} />
    </BasketContainer>
  );
};

export default Basket;
