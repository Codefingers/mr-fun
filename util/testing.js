import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={{
        colors: {
          body: '#838fa6',
          heading: '#ebedf2',
          accent: '#BA99FF',
          button: '#FFF',
        },
      }}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
