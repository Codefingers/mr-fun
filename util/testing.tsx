import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={theme}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
