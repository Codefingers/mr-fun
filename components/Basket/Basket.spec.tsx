import React from 'react';
import { render } from '../../util/testing';

import { Basket } from './Basket';

describe('<Basket />', () => {
  it('renders', () => {
    const { container } = render(<Basket />);
    expect(container).toMatchSnapshot();
  });
});
