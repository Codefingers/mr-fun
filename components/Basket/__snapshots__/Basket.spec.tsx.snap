// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basket /> renders 1`] = `
<div>
  <div
    class="Basket__BasketContainer-sc-kcuzmw-0 eRChUV"
  >
    <h2>
      Your Basket
    </h2>
    <p>
      You're currently looking at purchasing the following passes:
    </p>
    <div />
    <div
      class="Total__TotalContainer-sc-149mlyd-0 kkKrkF"
    >
      <h2>
        Total: 
        Â£0
      </h2>
    </div>
  </div>
</div>
`;
